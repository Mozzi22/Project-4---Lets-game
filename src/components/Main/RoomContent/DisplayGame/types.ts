export type TSingleSell = {
    id: number;
    isChecked: boolean | null;
    color: string | null;
    position: number;
}
