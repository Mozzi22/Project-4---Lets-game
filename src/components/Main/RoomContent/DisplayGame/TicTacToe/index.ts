import { connect } from 'react-redux';
import { getTicTacToeDesk } from 'src/Store/games/selectors';
import { do_step } from 'src/Store/games/actions';
// import { AplicationState } from 'src/Store/types';
import TicTacToe from './TicTacToe';

const mapStateToProps = (state) => ({
    ticTacToe: getTicTacToeDesk(state),
});

const mapDispatchToProps = {
  do_step,
};

export default connect(mapStateToProps, mapDispatchToProps)(TicTacToe);
