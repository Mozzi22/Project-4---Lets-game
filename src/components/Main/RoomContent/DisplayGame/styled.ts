import styled from 'styled-components';

export const StDisplayGame = styled.div`
    width: 80%;
    display: flex;
    align-content: center;
    align-items: center;
`;
