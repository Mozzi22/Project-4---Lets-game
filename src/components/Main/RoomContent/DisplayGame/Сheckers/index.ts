// import { connect } from 'react-redux';
// import { getCheckerDesk } from 'src/Store/games/selectors';
// import { ApplicationState } from 'src/Store/types';
import Checkers from './Checkers';

// const mapStateToProps = (state: ApplicationState) => ({
//   checker: getCheckerDesk(state),
// });

export default Checkers;
