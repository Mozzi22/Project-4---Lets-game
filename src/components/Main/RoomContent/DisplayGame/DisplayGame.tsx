import React from 'react';
import { StDisplayGame } from './styled';
import Checkers from './Ð¡heckers';
import TicTacToe from './TicTacToe';

const DisplayGame = ({ gameType }) => {
    switch (gameType) {
        case 'Checkers': return (
            <StDisplayGame>
                {/* <Turn/> */}
                <Checkers />
            </StDisplayGame>
        );
        case 'Tic-tac-toe': return (
            <StDisplayGame>
                {/* <Turn/> */}
                <TicTacToe />
            </StDisplayGame>
        );
        default: return (
            <StDisplayGame>
               <Welcome/>
            </StDisplayGame>
        );
    };
}

export default DisplayGame;