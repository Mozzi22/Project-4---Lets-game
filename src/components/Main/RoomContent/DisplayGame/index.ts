import { connect } from 'react-redux';
import { ApplicationState } from 'src/Store/types';
import { getActualRoomGameType } from 'src/Store/games/selectors';
import DisplayGame from './DisplayGame';

const mapStateToProps = store => ({ gameType: getActualRoomGameType(store) });

export default connect(mapStateToProps)(DisplayGame);
