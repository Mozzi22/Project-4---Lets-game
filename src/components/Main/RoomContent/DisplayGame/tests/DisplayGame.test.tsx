import React from 'react';
import DisplayGame from '../DisplayGame';
import { mount, shallow } from 'enzyme';

describe('DisplayGame', () => {
    it('Should match snapshot', () => {
        const component = shallow(<DisplayGame />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StDisplayGame', () => {
        const component = mount(<DisplayGame/>);
        expect(component.find('styled__StDisplayGame')).toHaveLength(1);
    });
    it('should render TicTacToe', () => {
        const component = mount(<DisplayGame/>);
        expect(component.find('TicTacToe')).toHaveLength(1);
    });
    it('should render Checkers', () => {
        const component = mount(<DisplayGame />);
        expect(component.find('Checkers')).toHaveLength(1);
    });
});