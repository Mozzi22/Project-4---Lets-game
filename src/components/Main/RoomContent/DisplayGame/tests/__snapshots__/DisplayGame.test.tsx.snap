// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayGame Should match snapshot 1`] = `"<div class=\\"styled__StDisplayGame-sc-d2zbbn-0 bZlcUH\\"><div class=\\"styled__StTicTacToe-sc-3am6n8-0 bMwVWP\\"><div id=\\"9\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">9</div><div id=\\"8\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">8</div><div id=\\"7\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">7</div><div id=\\"6\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">6</div><div id=\\"5\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">5</div><div id=\\"4\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">4</div><div id=\\"3\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">3</div><div id=\\"2\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">2</div><div id=\\"1\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 cKJSAA\\">1</div></div><div class=\\"styled__StCheckers-sc-1hqaub2-0 kNwXMw\\"><div id=\\"64\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">64</div><div id=\\"63\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">63</div><div id=\\"62\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">62</div><div id=\\"61\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">61</div><div id=\\"60\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">60</div><div id=\\"59\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">59</div><div id=\\"58\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">58</div><div id=\\"57\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">57</div><div id=\\"56\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">56</div><div id=\\"55\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">55</div><div id=\\"54\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">54</div><div id=\\"53\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">53</div><div id=\\"52\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">52</div><div id=\\"51\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">51</div><div id=\\"50\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">50</div><div id=\\"49\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">49</div><div id=\\"48\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">48</div><div id=\\"47\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">47</div><div id=\\"46\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">46</div><div id=\\"45\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">45</div><div id=\\"44\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">44</div><div id=\\"43\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">43</div><div id=\\"42\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">42</div><div id=\\"41\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">41</div><div id=\\"40\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">40</div><div id=\\"39\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">39</div><div id=\\"38\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">38</div><div id=\\"37\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">37</div><div id=\\"36\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">36</div><div id=\\"35\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">35</div><div id=\\"34\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">34</div><div id=\\"33\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">33</div><div id=\\"32\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">32</div><div id=\\"31\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">31</div><div id=\\"30\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">30</div><div id=\\"29\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">29</div><div id=\\"28\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">28</div><div id=\\"27\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">27</div><div id=\\"26\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">26</div><div id=\\"25\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">25</div><div id=\\"24\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">24</div><div id=\\"23\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">23</div><div id=\\"22\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">22</div><div id=\\"21\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">21</div><div id=\\"20\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">20</div><div id=\\"19\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">19</div><div id=\\"18\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">18</div><div id=\\"17\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">17</div><div id=\\"16\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">16</div><div id=\\"15\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">15</div><div id=\\"14\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">14</div><div id=\\"13\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">13</div><div id=\\"12\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">12</div><div id=\\"11\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">11</div><div id=\\"10\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">10</div><div id=\\"9\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">9</div><div id=\\"8\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">8</div><div id=\\"7\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">7</div><div id=\\"6\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">6</div><div id=\\"5\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">5</div><div id=\\"4\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">4</div><div id=\\"3\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">3</div><div id=\\"2\\" style=\\"background:white\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">2</div><div id=\\"1\\" style=\\"background:black\\" class=\\"styled__StSingleSell-sc-1cetqro-0 bticym\\">1</div></div></div>"`;
